framework/include/variables/MooseVariableBase.h

```diff
@@ -172,6 +172,13 @@ class MooseVariableBase : public MooseObject,
    */
   bool isArray() const { return _is_array; }
 
+  /**
+   * Request that quadrature point data be pre-computed. This is always done for finite element
+   * variables, but finite volume variables may not pre-compute quadrature point data unless this
+   * API is called
+   */
+  virtual void requireQpComputations() {}
```

Comment:
This is cool! Just a quick question: in what cases do we need this for FV at the moment? I was thinking about this lately, if we didn't need thee evaluations we could get rid of most of the `MooseVariableFVData`, right? Or maybe I am wrong here. I suppose when we evaluate with quadrature points we need these.

Reply:
The fully compressible finite volume kernels are currently using qp indexing